server/src/main/java/org/opensearch/rest/action/cat/RestIndicesAction.java

```diff
@@ -205,6 +218,91 @@ public void onFailure(final Exception e) {
         };
     }
 
+    public RestChannelConsumer doPaginatedCatRequest(
```

Comment:
Should this method be declared as abstract in `AbstractCatAction` ?

Reply:
Have introduced a new doListRequest method now, given that new RestListAction had to be added.